src/libraries/Microsoft.PowerFx.Interpreter/UDFHelper.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    internal class UDFHelper
+    {
+        public class LazyCheck
+        {
+            internal string _expressionText;
+            internal RecordType _parameterType;
+            internal ParserOptions _options;
+            internal Engine _engine;
```

Comment:
>_engine [](http://example.com/codeflow?start=28&length=7)

naming nit: internals should start with a capital letter. 
_ is only for privates.

Reply:
Actually, these should probably all be privates

Reply:
+1 to using privates instead of internal. Especially if it's 'private readonly' and only set in ctor.

Reply:
Done